org.knime.knip.ilastik/src/org/knime/knip/ilastik/nodes/headless/IlastikHeadlessNodeModel.java

```diff
@@ -398,12 +412,23 @@ private boolean runIlastik(final String tmpDirPath, final List<String> inFiles)
      * @param in input stream
      * @throws IOException
      */
-    static void writeToKnimeConsole(final InputStream in) throws IOException {
-        BufferedReader bis = new BufferedReader(new InputStreamReader(in, Charset.defaultCharset()));
-        String line;
-        while ((line = bis.readLine()) != null) {
-            KNIPGateway.log().info(line);
-        }
+    static void writeToKnimeConsole(final InputStream in) {
+
+        new Thread(
```

Comment:
I wouldn't create a new thread for each write request. this will really harm performance if the gc has to take care about cleaning-up this mess  ;-) Can you just create two threads which you reuse?